### PR TITLE
Conversion to umbrella project and other improvements

### PR DESCRIPTION
**Pull request by kevinbader**

### CODE REVIEW

**Comment by kevinbader** [comment]
@mmacai with 1.1.0 out the door, it's time to get this merged into mainline and iterate from there

File: apps/rig_auth/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+## Installation
+
+If [available in Hex](https://hex.pm/docs/publish), the package can be installed
+by adding `rig_auth` to your list of dependencies in `mix.exs`:
+
+```elixir
+def deps do
+  [
+    {:rig_auth, "~> 0.1.0"}
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply apps/rig_auth/README.md]: I'd say we remove this altogether

File: apps/rig_auth/test/blacklist_test.exs [hunk 2]
```diff
@@ -69,49 +74,46 @@ defmodule Rig.BlacklistTest do
...
-          else
-            Agent.update(agent, fn
-              list -> [{jti, %{expiry: expiry}} | list]
-            end)
-            {:ok, 'some_phx_ref'}
+
+    tracker =
+      Stubr.stub!(
+        [
+          # @callback track(jti: String.t, expiry: Timex.DateTime.t) :: {:ok, String.t}
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply apps/rig_auth/test/blacklist_test.exs]: hm that's meant as documentation as to what the mock should provide. But it's not checked by the compiler, so perhaps should be removed, not sure

File: apps/rig_inbound_gateway/config/dev.exs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+# watchers to your application. For example, we use it
+# with brunch.io to recompile .js and .css sources.
+config :rig, RigInboundGatewayWeb.Endpoint,
+  # debug_errors: true, # Uncomment to see full error descriptions in API as HTML
+  env: :dev,
+  check_origin: false,
+  watchers: []
+
+# Do not include metadata nor timestamps in development logs TODO why?
+#config :logger, :console, format: "[$level] $message\n"
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply apps/rig_inbound_gateway/config/dev.exs]: jep

File: apps/rig_api/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# RigApi
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply apps/rig_api/README.md]: I'll keep the README files but only include a link to the developer's guide

File: apps/rig_inbound_gateway/lib/rig_inbound_gateway_web/presence/channel.ex [hunk 5]
```diff
@@ -77,6 +77,20 @@ defmodule RigWeb.Presence.Channel do
     end
   end
 
+  @doc """
+  Sends off outgoing messages.
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply apps/rig_inbound_gateway/lib/rig_inbound_gateway_web/presence/channel.ex]: I still think it's rather strange

File: apps/rig_inbound_gateway/mix.exs [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  # Type `mix help deps` for examples and options.
+  defp deps do
+    [
+      {:rig, in_umbrella: true},
+      {:rig_auth, in_umbrella: true},
+      {:phoenix, "~> 1.3.0"},
+      {:phoenix_pubsub, "~> 1.0"},
+      {:cowboy, "~> 1.0"},
+      {:httpoison, "~> 0.13.0"},
+      # {:joken, "~> 1.4"},
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply apps/rig_inbound_gateway/mix.exs]: done

File: apps/rig_outbound_gateway/README.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+configurable function invocation; this should make for clear touch points to
+`rig_inbound_gateway`.
+
+Currently supported sources:
+
+- Kafka (topic with JSON payload)
+
+Work in progress:
+
+- HTTP API
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply apps/rig_outbound_gateway/README.md]: removed that

File: apps/rig_outbound_gateway/mix.exs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  defp deps do
+    [
+      {:rig, in_umbrella: true},
+      # Read and use application configuration from environment variables:
+      {:confex, "~> 3.3"},
+      # Apache Kafka Erlang client library:
+      {:brod, "~> 3.3"},
+      # JSON parser:
+      {:poison, "~> 2.0 or ~> 3.0"},
+      # HTTP server:
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply apps/rig_outbound_gateway/mix.exs]: done

File: config/config.exs [hunk 9]
```diff
@@ -1,58 +1,32 @@
...
 
+metadata = [
+  # phoenix_metadata
+  :request_id,
+  # kafka_metadata
+  :raw,
+  :topic,
+  :partition,
+  :offset,
+  # aws_sqs_metadata
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply config/config.exs]: true

File: config/dev.exs [hunk 10]
```diff
@@ -1,18 +1,6 @@
...
-# with brunch.io to recompile .js and .css sources.
-config :rig, RigWeb.Endpoint,
-  # debug_errors: true, # Uncomment to see full error descriptions in API as HTML
-  env: :dev,
-  check_origin: false,
-  watchers: []
-
-# Do not include metadata nor timestamps in development logs TODO why?
+# Do not include metadata nor timestamps in development logs
 #config :logger, :console, format: "[$level] $message\n"
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply config/dev.exs]: done

File: version [hunk 11]
```diff
@@ -0,0 +1,4 @@
+%{
+  rig: "2.0.0-dev",
+  elixir: "~> 1.5",
```
[the remark this answers is not in the document]
Comment (kevinbader) [reply version]: okay

**Comment by mmacai** [comment]
- Missing `COPY apps/rig_auth/config /opt/sites/rig/apps/rig_auth/config` => leads to error during `docker build ...`

- Since `RUN mix release.init` step was removed from `Dockerfile`, it should be removed from `.gitignore` and run manually `MIX_ENV=prod mix release.init`

- even the fixes above, still getting error at the end of `docker build ...`
```
==> Assembling release..
==> Building release reactive_interaction_gateway:0.1.0 using environment prod
==> Including ERTS 9.2 from /usr/local/lib/erlang/erts-9.2
==> Packaging release..
==> Release packaging failed due to errors:
    Cannot add file sys.config to tar file - [{error,{25,erl_parse,["syntax error before: ",["Fun"]]}}]
```

**Comment by mmacai** [comment]
Tested with channels example and with some minor changes in example worked for both ws and sse.

**Pull request merged**